src/timedate/timedated.c

```diff
@@ -545,7 +546,12 @@ static int unit_enable_or_disable(UnitStatusInfo *u, sd_bus *bus, sd_bus_error *
         if (r < 0)
                 return r;
 
-        r = bus_call_method(bus, bus_systemd_mgr, "Reload", error, NULL, NULL);
+        /* Reloading the daemon may take long, hence set a longer timeout here */
+        r = bus_message_new_method_call(bus, &m, bus_systemd_mgr, "Reload");
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = sd_bus_call(bus, m, DAEMON_RELOAD_TIMEOUT_SEC, error, NULL);
```

Comment:
so we added a helper for this case some time ago which uses the right time-out, it's available in `bus-unit-util.[ch]`. Please use that here, i.e. replace the whole bus_call_method() with a simple call to bus_service_manager_reload().